homeassistant/components/aidot/coordinator.py

```diff
@@ -0,0 +1,168 @@
+"""Coordinator for Aidot."""
+
+import logging
+from typing import Any
+
+from aidot.client import AidotClient
+from aidot.const import (
+    CONF_ACCESS_TOKEN,
+    CONF_AES_KEY,
+    CONF_DEVICE_LIST,
+    CONF_ID,
+    CONF_LOGIN_INFO,
+    CONF_TYPE,
+)
+from aidot.device_client import DeviceClient, DeviceInformation, DeviceStatusData
+from aidot.exceptions import AidotAuthFailed, AidotNotLogin, AidotUserOrPassIncorrect
+
+from homeassistant.components.sensor import timedelta
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+
+type AidotConfigEntry = ConfigEntry[AidotDeviceManagerCoordinator]
+_LOGGER = logging.getLogger(__name__)
+
+
+class AidotDeviceUpdateCoordinator(DataUpdateCoordinator[DeviceStatusData]):
+    """Class to manage Aidot data."""
+
+    device: dict[str, Any]
+    ip_address: str
+    device_client: DeviceClient
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config_entry: AidotConfigEntry,
+        device_client: DeviceClient,
+    ) -> None:
+        """Initialize coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            config_entry=config_entry,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=1),
+        )
+        self.device_client = device_client
+        # self.hass.loop.create_task(device_client.ping_task())
+        self.identifier = config_entry.entry_id
+
+    async def _async_setup(self) -> None:
+        """Set up the coordinator.
+
+        Can be overwritten by integrations to load data or resources
+        only once during the first refresh.
+        """
+        try:
+            await self.device_client.async_login()
+        except AidotUserOrPassIncorrect as error:
+            raise ConfigEntryError from error
+
+    async def _async_update_data(self) -> DeviceStatusData:
+        """Update data async."""
+        try:
+            status = await self.device_client.read_status()
+        except AidotNotLogin:
+            await self.device_client.async_login()
+            return DeviceStatusData()
+        return status
+
+    @property
+    def device_info(self) -> DeviceInformation:
+        """Device information."""
+        return self.device_client.info
+
+
+class AidotDeviceManagerCoordinator(
+    DataUpdateCoordinator[dict[str, AidotDeviceUpdateCoordinator]]
```

Comment:
```suggestion
    DataUpdateCoordinator[None]
```
we don't have entities listening on those as data, so no point to return them here